packages/block-library/src/legacy-widget/block.json

```diff
@@ -17,6 +17,10 @@
 		"instance": {
 			"type": "object",
 			"default": null
+		},
+		"isWide": {
```

Comment:
I'm not 100% sure this should be an attribute because it's not "content" per se. For example, if a Legacy Widget block is in a post, and `isWide` is toggled somehow, should that create an undo/redo level in the block editor?

Legacy Widget has `idBase` available to it. Could we have the Legacy Widget block itself decide whether or not it should be wide or not by accessing `wp.customize.Widgets.data.availableWidgets` from `packages/block-library/src/legacy-widget/edit/index.js`? The block already heavily relies on older WordPress APIs so I don't think it's a problem to do this. It will just have to be careful to account for the case that `wp.customize` is undefined.

Reply:
Another option would be to add `is_wide` to the `/widget-types` REST API endpoint.

Reply:
Not sure I follow, if we added `is_wide` to the endpoint wouldn't we end up having to add it to the block attributes too?

Reply:
So trying to get `isWide` directly inside the Legacy Widget block through `wp.customize...` works well, and fixes an issue I was having with widgets not having the value set when they're first added to the screen. But now I have to somehow figure out if the block is rendering in the Customizer or not, because we only want to show the popover in the Customizer, not in the Widgets screen. Aside from the general yuckiness of having a block be aware of where it's being rendered, is there any way of doing this that doesn't involve checking the current URL?

Or... I dunno. `isWide` is as likely to change as `idBase`. I don't think it's _too_ awful to have it in attributes. Or is there another way?

Reply:
There's also the option of using a `BlockEdit` filter to help implement this. That filter will have access to all the attributes for the block and would be implemented only in the customizer.

An extreme option—not sure if doing something like putting the whole `BlockEdit` in a popover using the filter would work:
```jsx
{ isWide && (
  <Popover>
    <BlockEdit { ...props } />
  </Popover>
) }
```

Seems a little crazy, but would keep the legacy widget block pure. The lighter alternative might be passing an `isWide` prop down to `BlockEdit`.

The other option I could think of is to implement the feature conditionally based on the availability of the `wp.customize` api, which isn't defined in the standalone editor. That would break if it ever were available.

Reply:
Ha, turns out I was doing something wrong and `isWide` was always true, so that's why popovers were rendering in the Widgets screen 😂 

Ooooh BlockEdit, good tip! I'll look into it.

Reply:
Ok so putting BlockEdit inside the Popover won't work because we only want to render a part of the form in the popover, not the whole edit. But I added a filter to pass in `isWide` which seems much cleaner.

Reply:
Sorry, I went MIA 😅 

> Not sure I follow, if we added is_wide to the endpoint wouldn't we end up having to add it to the block attributes too?

No, because we access the `widget-types` endpoint from within `Edit`.

https://github.com/WordPress/gutenberg/blob/cc41febc1d0b6c753bab752d3e68fbe27f1a5399/packages/block-library/src/legacy-widget/edit/index.js#L95

> But now I have to somehow figure out if the block is rendering in the Customizer or not, because we only want to show the popover in the Customizer, not in the Widgets screen. Aside from the general yuckiness of having a block be aware of where it's being rendered, is there any way of doing this that doesn't involve checking the current URL?

`wp.customize` will be undefined if the block is not in the Customizer.

Reply:
I looked into how `is_wide` is implemented in Core and I don't think it makes sense to add `is_wide` to the REST API because the logic is so specific to the Customizer. 

So I think your current approach in this PR is the right one.